### PR TITLE
Always log request and response in httpclient Do()

### PR DESCRIPTION
**Pull request by nicolaasuni-vonage**

### CODE REVIEW

File: pkg/httpclient/client.go [hunk 1]
```diff
@@ -77,28 +77,29 @@ func (c *Client) Do(r *http.Request) (resp *http.Response, err error) {
...
+		}
+
+		l = l.With(
+			zap.String(c.logPrefix+"traceid", reqID),
+			zap.String(c.logPrefix+"request_method", r.Method),
+			zap.String(c.logPrefix+"request_path", r.URL.Path),
+			zap.String(c.logPrefix+"request_query", r.URL.RawQuery),
+			zap.String(c.logPrefix+"request_uri", r.RequestURI),
+			zap.String(c.logPrefix+"request", c.redactFn(string(reqDump))),
+			zap.String(c.logPrefix+"response", c.redactFn(string(respDump))),
```
Comment (conves) [inline pkg/httpclient/client.go:100]: Isn't this a behavior change? Before, these fields were logged no matter what, now they are only logged if the debug is activated.

File: pkg/httpclient/client.go [hunk 2]
```diff
@@ -86,19 +88,26 @@ func (c *Client) Do(r *http.Request) (resp *http.Response, err error) {
...
 
 	resp, err = c.client.Do(r)
-	if err != nil {
-		return nil, err //nolint:wrapcheck
-	}
 
 	if debug {
-		respDump, _ := httputil.DumpResponse(resp, true)
-		l = l.With(zap.String(c.logPrefix+"response", c.redactFn(string(respDump))))
+		if resp != nil {
```
Comment (conves) [inline pkg/httpclient/client.go:102]: Is it possible to consolidate these checks?

File: pkg/httpclient/client_test.go [hunk 3]
```diff
@@ -73,24 +77,61 @@ func TestClient_Do(t *testing.T) {
...
+
+// MemorySink implements zap.Sink by writing all messages to a buffer.
+type MemorySink struct {
+	*bytes.Buffer
+}
+
+// Implement Close and Sync as no-ops to satisfy the interface. The Write
+// method is provided by the embedded buffer.
+
+func (s *MemorySink) Close() error { return nil }
```
Comment (conves) [inline pkg/httpclient/client_test.go:136]: Would it help the GC if we put a https://pkg.go.dev/bytes#Buffer.Reset call here?

File: pkg/httpclient/client_test.go [hunk 4]
```diff
@@ -73,24 +77,61 @@ func TestClient_Do(t *testing.T) {
...
+	require.NoError(t, err)
 	require.Equal(t, body, responseBody)
+
+	// check logs
+	out := sink.String()
+	require.NotEmpty(t, out, "captured log output")
+	require.Contains(t, out, `"traceid"`)
+	require.Contains(t, out, `"request":"GET / HTTP/1.1`)
+	require.Contains(t, out, `"response":"HTTP/1.1 200 OK`)
+	require.Contains(t, out, bodyStr)
```
Comment (conves) [inline pkg/httpclient/client_test.go:125]: If possible, would it help to adjust this on to table testing, so we can also test the `logPrefix` functionality, which is not currently covered?

File: pkg/httpclient/client_test.go (hunk 3 above)
Comment (conves) [reply pkg/httpclient/client_test.go:180]: I'm only thinking that the `zap.Logger` is going to actually write on that byte buffer, and it wouldn't have hurt to clear it up. But there are tests so we can ignore it.

**Review by conves — APPROVED** [review]

**Review by anup-limbu — APPROVED** [review]

**Pull request merged**